lib/plugins/openaps.js

```diff
@@ -0,0 +1,168 @@
+
+    var info = [];
+
+    var recent = moment(sbx.time).subtract(15, 'minutes');
+    var suggested = prop.openaps && prop.openaps.suggested;
+    var suggestedTime = suggested && moment(suggested.timestamp);
+    var enacted = prop.openaps && prop.openaps.enacted;
+    var enactedTime = enacted && moment(enacted.timestamp);
+    var lastLoopTime = suggestedTime && enactedTime ? moment.max(suggestedTime, enactedTime) : enactedTime || suggestedTime;
+
+    function valueString (prefix, value) {
+      return value ? prefix + value : '';
+    }
+
+    function addedSuggestion() {
+      info.push({
+        label: timeAt(false, sbx) + timeFormat(suggestedTime, sbx)
+        , value: valueString('BG: ', suggested.bg) + valueString(', ', suggested.reason)
+      });
+    }
+
+    if ('enacted' === prop.status.code) {
+      info.push({label: 'Status' , value: 'Enacted'});
+      if (suggestedTime.isAfter(enactedTime)) {
+        addedSuggestion();
+      }
+      var canceled = enacted.rate === 0 && enacted.duration === 0;
+      info.push({
+        label: timeAt(false, sbx) + timeFormat(enactedTime, sbx)
+        , value: [
+          valueString('BG: ', enacted.bg)
+          , ', <b>Temp Basal' + (canceled ? ' Canceled' : ' Started') + '</b>'
+          , canceled ? '' : ' ' + enacted.rate.toFixed(2) + ' for ' + enacted.duration + 'm'
+          , valueString(', ', enacted.reason)
+        ].join('')
+      });
+    } else if ('looping' === prop.status.code) {
+      info.push({
+        label: 'Status'
+        , value: 'Looping'
+      });
+      addedSuggestion();
+    } else if ('waiting' === prop.status.code) {
+      info.push({
+        label: 'Waiting'
+        , value: lastLoopTime ? 'last loop: ' + timeFormat(lastLoopTime, sbx) : ''
+      });
+    } else if ('warning' === prop.status.code) {
+      info.push({
+        label: 'Warning'
+        , value: 'No recent status updates'
+      });
+    }
+
+    if (prop.pump && prop.pump.clock) {
+      var pumpTime = moment(prop.pump.clock);
+      if (pumpTime.isAfter(recent)) {
+        info.push({label: 'Last Pump Clock', value: pumpTime.format('LT')});
+      }
+    }
+
+    if (prop.openaps && prop.openaps.iob) {
+      var iobTime = moment(prop.openaps.iob.timestamp);
+      if (iobTime.isAfter(recent)) {
+        info.push({label: 'IOB', value: sbx.roundInsulinForDisplayFormat(prop.openaps.iob.iob) + 'U' + timeAt(true, sbx) + timeFormat(iobTime, sbx)});
+        info.push({label: 'Bolus IOB', value: sbx.roundInsulinForDisplayFormat(prop.openaps.iob.bolusiob) + 'U'});
+        info.push({label: 'Insulin Activity', value: prop.openaps.iob.activity.toFixed(2) + ' ' + sbx.unitsLabel + '/5m'});
+      }
+    }
+
+    sbx.pluginBase.updatePillText(openaps, {
+      value: timeFormat(prop.status.when, sbx)
+      , label: 'OpenAPS ' + prop.status.symbol
+      , info: info
+    });
+  };
+
+  return openaps;
+
+}
+
+function checkLoopStatus (openapsStatus, sbx)  {
```

Comment:
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: [This function's cyclomatic complexity is too high. (8)](https://www.codacy.com/app/Nightscout/cgm-remote-monitor/file/1445666522/issues/source?bid=2452379&fileBranchId=2989096#l113)